measurement/plugins/ip_route/measurements.py

```diff
@@ -0,0 +1,116 @@
+Traceroute measurement is provided by the "scapy" library. Note that this library needs route privileges to create
+raw sockets, or to have the CAP_NET_RAW capability set.
+
+"""
+import socket
+import validators
+from validators import ValidationFailure
+
+import scapy
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.plugins.ip_route.results import IPRouteMeasurementResult
+from measurement.plugins.latency.measurements import LatencyMeasurement
+
+ROUTE_ERRORS = {
+    "route-err": "iproute encountered an unknown error",
+    "route-address": "iproute failed to find socket address",
+    "route-permission": "iproute does not have permission to create a raw socket",
+}
+
+
+class IPRouteMeasurement(BaseMeasurement):
+    def __init__(self, id, hosts, route_timeout=10, count=4):
+        super(IPRouteMeasurement, self).__init__(id=id)
+
+        if len(hosts) < 1:
+            raise ValueError("At least one url must be provided.")
+        for host in hosts:
+            validated_url = validators.domain(host)
+            if isinstance(validated_url, ValidationFailure):
+                raise ValueError("`{host}` is not a valid host".format(host=host))
+
+        if count < 0:
+            raise ValueError(
+                "A value of {count} was provided for the number of pings. This must be a positive "
+                "integer or `0` to turn off the ping.".format(count=count)
+            )
+
+        self.id = id
+        self.hosts = hosts
+        self.route_timeout = route_timeout
+        self.count = count
+
+    def measure(self):
+        initial_latency_results = self._find_least_latent_host(self.hosts)
+        least_latent_host = initial_latency_results[0][0]
+        results = [self._get_traceroute_result(least_latent_host)]
+        if self.count > 0:
+            latency_measurement = LatencyMeasurement(
+                self.id, least_latent_host, count=self.count
+            )
+            results.append(latency_measurement.measure()[0])
+        results.extend([res for _, res in initial_latency_results])
+        print(results)
+        return results
+
+    def _find_least_latent_host(self, hosts):
+        """
+        Performs a latency test for each specified host
+        Returns a sorted list of LatencyResults, sorted by average latency
+        """
+        initial_latency_results = []
+        for host in hosts:
+            print("host: ", host)
+            latency_measurement = LatencyMeasurement(self.id, host, count=2)
+            initial_latency_results.append((host, latency_measurement.measure()[0]))
+        print("initial latency results: ", initial_latency_results)
+        return sorted(
+            initial_latency_results,
+            key=lambda x: (x[1].average_latency is None, x[1].average_latency),
+        )
+
+    def _get_traceroute_result(self, host):
+        try:
+            traceroute_out = scapy.layers.inet.traceroute(host)
```

Comment:
This is throwing the error below for me. I've confirmed that I have scapy `2.4.2` installed in my venv.


```
In [8]: r = irm.measure()                                                                                                                                                                                          
host:  www.google.com
host:  www.facebook.com
initial latency results:  [('www.google.com', LatencyMeasurementResult(id='3c31cd4f-721b-49f9-923c-5a932ff04d81', errors=[], host='www.google.com', minimum_latency=16.189, average_latency=16.556, maximum_latency=16.923, median_deviation=0.367, packets_transmitted=2, packets_received=2, packets_lost=0.0, packets_lost_unit=<RatioUnit.percentage: '%'>, time=1002.0, time_unit=<TimeUnit.millisecond: 'ms'>)), ('www.facebook.com', LatencyMeasurementResult(id='3c31cd4f-721b-49f9-923c-5a932ff04d81', errors=[], host='www.facebook.com', minimum_latency=15.679, average_latency=16.088, maximum_latency=16.498, median_deviation=0.409, packets_transmitted=2, packets_received=2, packets_lost=0.0, packets_lost_unit=<RatioUnit.percentage: '%'>, time=1002.0, time_unit=<TimeUnit.millisecond: 'ms'>))]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-8-c2b4d0383637> in <module>
----> 1 r = irm.measure()

~/dev/honestybox/honestybox-measurement/measurement/plugins/ip_route/measurements.py in measure(self)
     53         initial_latency_results = self._find_least_latent_host(self.hosts)
     54         least_latent_host = initial_latency_results[0][0]
---> 55         results = [self._get_traceroute_result(least_latent_host)]
     56         if self.count > 0:
     57             latency_measurement = LatencyMeasurement(

~/dev/honestybox/honestybox-measurement/measurement/plugins/ip_route/measurements.py in _get_traceroute_result(self, host)
     81     def _get_traceroute_result(self, host):
     82         try:
---> 83             traceroute_out = scapy.layers.inet.traceroute(host)
     84             traceroute_trace = traceroute_out[0].get_trace()
     85             ip = list(traceroute_trace.keys())[0]

AttributeError: module 'scapy' has no attribute 'layers'
```